packages/core/src/sync-realtime/index.ts

```diff
@@ -282,11 +288,56 @@ export const createRealtimeSync = (
         }' blocks from ${hexToNumber(finalizedBlock.number) + 1} to ${pendingFinalizedBlock.number}`,
       });
 
+      const finalizedBlocks = localChain.filter(
+        (lb) =>
+          hexToNumber(lb.number) <= hexToNumber(pendingFinalizedBlock.number),
+      );
+
       localChain = localChain.filter(
         (lb) =>
           hexToNumber(lb.number) > hexToNumber(pendingFinalizedBlock.number),
       );
 
+      // add child address from newly finalized blocks to `finalizedChildAddresses`
+      for (const filter of factories) {
```

Comment:
Nit, but I'd find this more readable if the `factories` loop came after the `if (factoryLogs !== undefined)` check